cluster-install/install.sh

```diff
@@ -1004,10 +1040,9 @@ get_registry_credentials_from_pull_secret() {
     # 6. Get the first matching entry's `.value.auth`, which would be `B64_ENCODED_USERNAME_PASSWORD` here
     # 7. The `B64_ENCODED_USERNAME_PASSWORD` can then be passed to curl via
     # `-H 'Authorization: Basic B64_ENCODED_USERNAME_PASSWORD'` without having to decode it first.
-    local -r contents="$(kubectl get secret "$pull_secret" -n "$namespace" -o json 2> /dev/null)"
+    local -r contents="$(k8s_get_resource "secret/$pull_secret" "$namespace" "json")"
```

Comment:
Added a two new functions that are a lot of the changes in this PR:
- `k8s_get_resource`
- `k8s_resource_exists`
Both are pretty self-explanatory, but the main reason for their existence is to handle the separation of different errors (not found vs timeout/connection/etc.) we might encounter when trying to run these kubectl commands. 

A good example for why we need this is how we detect whether Trident is installed or not. Right now, any error will essentially lead to us detecting that "trident is not installed", even if the error was due to a timeout or a temporary connection blip. This causes the script to continue execution, assuming trident is not installed, which then leads to it reinstalling Trident and potentially stomping all over a user's Trident install.

You'll get to see the actual function def just a bit below, where I'll point out the different errors being handled.